### ISSUE TITLE
Custom item template

### ISSUE THREAD

**Issue by RomainTT** [jacebrowning/doorstop#330]
Possibly embraces #70 and #319, with a little bit more.

It would be nice being able to write item templates like this:

```yaml
custom_field: custom_default_value
text: |
    Default text that appears in each new item.
```

And tell a document to use this template in `.doorstop.yml`:

```yaml
settings:
  digits: 3
  prefix: SRS
  sep: ''
  template: my_template.yml
```

Then, when I create a new item in SRS it has immediatly the content of the template:

```yaml
active: true
derived: false
level: 1.0
links: []
normative: true
ref: ''
reviewed: None
custom_field: custom_default_value
text: |
    Default text that appears in each new item.
```

Personally, I would need this to include some mandatory fields in items that are describing tests. And I want all my items to have the same fields with the same default value.

Meanwhile, I don’t know where should be located the template files, maybe anywhere since doorstop can search the whole depository?

**Comment by jacebrowning** [comment]
Could the default fields be a part of `.doorstop.yml`? Maybe there could be a new `defaults` top-level dictionary?

**Comment by RomainTT** [comment]
I was thinking about this too. But let’s say I want to apply a common template to many documents, because for a specific project I am writing requirements with a number of custom fields, and these requirements are located in different documents. I would be forced to copy-paste my `defaults` dictionary to every `.doorstop.yml`. That could work for a few fields, but it becomes very annoying when managing like 10 fields for instance.

**Comment by TingForGitHub** [comment]
Due to ECSS standards, we need to add some more keys such as "Type", "Verification Method", etc. into the .yml files for requirements. But are custom keys tracked by the fingerprints and the review? If not, will this feature be considered to be added to a newer version of Doorstop?

We also want to have the custom keys with default value to take effect on every requirement .yml file once we define the custom keys.

**Comment by jacebrowning** [comment]
Custom keys are not currently included in the fingerprint: 

https://github.com/jacebrowning/doorstop/blob/569a24eb42423708adeed4e17a19188a8e733e56/doorstop/core/item.py#L862-L864

I would be happy to review a PR that includes custom keys in the fingerprint.

**Comment by sebhub** [comment]
I think this issue asks for two things:

1. A way to initialize extended attributes with document-specific default values. This is #70 

2. A way to include other YAML files in .doorstop.yml.

The second thing could be handled with something like this:
```yaml
include: absolute/or/relative/path/to/some.yml
```
Or:
```yaml
include:
- absolute/or/relative/path/to/some.yml
- absolute/or/relative/path/to/other.yml
```
The question is, if this should work recursively. The included YAML files could fill up a `defaults` set which initializes extended attributes (or any attribute?).

A follow up feature could add initialization via custom Python code.

**Comment by jacebrowning** [comment]
There's an example [here](https://stackoverflow.com/a/9577670/429533) of defining a custom `!include` constructor that might work for this.

**Comment by sebhub** [comment]
I added a new issue for the include feature: #367. The default value part was implemented by commit 6c91fab39f9a2d5b1e46e7276e89f4e803cdde8c.
